src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1358,6 +1358,170 @@ if rex then
   -- Backwards compatibility
   checho = cecho
 
+  -- table to facilitate converting color names to ansi escapes
+  local ctable =
+  {
+    black = "0",
+    red = "1",
+    green = "2",
+    yellow = "3",
+    blue = "4",
+    magenta = "5",
+    cyan = "6",
+    white = "7",
+    light_black = "8",
+    light_red = "9",
+    light_green = "10",
+    light_yellow = "11",
+    light_blue = "12",
+    light_magenta = "13",
+    light_cyan = "14",
+    light_white = "15",
+    lightBlack = "8",
+    lightRed = "9",
+    lightGreen = "10",
+    lightYellow = "11",
+    lightBlue = "12",
+    lightMagenta = "13",
+    lightCyan = "14",
+    lightWhite = "15",
+  }
+  for i = 0, 255 do
+    local key = tostring(i)
+    ctable[key] = key
+    ctable["ansi_" .. key] = key
+  end
+
+  -- take a color name and turn it into an ANSI escape string
+  local function colorToAnsi(colorName)
+    local result = ""
+    local cols = colorName:split(":")
+    local fore = cols[1]
+    local back = cols[2]
+    if fore ~= "" then
+      if fore == "r" or fore == "reset" then
+        result = result .. "\27[39;49m"
+      else
+        local colorNumber = ctable[fore]
+        if colorNumber then
+          result = string.format("%s\27[38;5;%sm", result, colorNumber)
+        end
+      end
+    end
+    if back then
+      local colorNumber = ctable[back]
+      result = string.format("%s\27[48;5;%sm", result, colorNumber)
+    end
+    return result
+  end
+
+  -- converts decho color information to ansi escape sequences
+  local function rgbToAnsi(rgb)
+    local result = ""
+    local cols = rgb:split(":")
+    local fore = cols[1]
+    local back = cols[2]
+    if fore ~= "" then
+      local components = fore:split(",")
+      result = string.format("%s\27[38;2;%s;%s;%sm", result, components[1] or "0", components[2] or "0", components[3] or "0")
```

Comment:
Please use the more correct *semi-colon* separated form (the semi-colon is a reserved character in ANSI ESC sequences and separating element in a sequence of ***sub**-parameters* seems to be what it is was reserved for!) - and also remember the (default) colour Id parameter that follows the `2`... - i.e. this one should become (and the other three need to be revised similarly), IMHO:
```lua
       result = string.format("%s\27[38:2::%s:%s:%sm", result, components[1] or "0", components[2] or "0", components[3] or "0")
```
so that it is more reliable...

Reply:
the semicolon is listed as a valid separate and clearly works. I'm not unwilling to change it but I'm curious why it is the : is somehow more correct? 
What do we actually gain by changing it up if Mudlet's terminal doesn't see it any differently?
Since only Mudlet will ever see it, how exactly is it meant to be unreliable unless Mudlet itself is unreliable?

Reply:
in testing, the : separator works for everything except the reset which doesn't wanna work as 39:49, only as 39;49

Given that, I'd rather use the semicolon everywhere if it functions, rather than waffling between them

Reply:
Semicolon seems fine to me as well, it's what I've seen most game servers use. This doesn't seem all that vital.

Reply:
The reason for using colons is that the semicolon is used to *separate **elements*** in a SGR sequence but the ones that *begin* with `38` or `48` can have a variable number of **sub-elements** values following them and for the ones with the first sub-element being `2`, `3` or `4` the sub-elements following that number are **all** optional - and if they are separated with semi-colons it is ***IMPOSSIBLE*** to determine with 100% confidence what the meaning is.

E.g. `<SGR> 38 : 2 : : 255 : 255 ; 1 m` is unambiguously a **bold** yellow with RGB values of (255,255,0) but `<SGR> 38 ; 2 ; 255 ; 255 ; 1 m` could be parsed as a non-bold red with RGB values of (255,1,0) {ignoring the colour space id of 255, and then taking the second 255 as the red component, the 1 as the green and then setting the omitted blue parameter to the default of 0 - this is a somewhat artificial example but it sort of starts to show the difficulty in parsing the sequence of numbers...

Reply:
In practice, this is not happening.

Reply:
> in testing, the : separator works for everything except the reset which doesn't wanna work as 39:49, only as 39;49

That is good because the `39` and `49` are separate SGR elements - so need to be separated with the element separator `;` only the `38` and `48` elements have *sub-elements*. (The specification uses *parameter* and *sub-parameter* but I forgot to use those terms in past messages...)

Reply:
From wikipedia's article on 24bit color in ANSI it seems the : separated one is a -less- supported option, and the ; delimited version I went with is both valid and more common. Now, wikipedia is hardly the final arbiter of truth it definitely seems to indicate my form is perfectly valid.

```
As "true color" graphic cards with 16 to 24 bits of color became common, Xterm,[19] KDE's Konsole,[37] iTerm, as well as all libvte based terminals[38] (including GNOME Terminal) support 24-bit foreground and background color setting.[better source needed][39]

ESC[ 38;2;⟨r⟩;⟨g⟩;⟨b⟩ m Select RGB foreground color
ESC[ 48;2;⟨r⟩;⟨g⟩;⟨b⟩ m Select RGB background color
The ITU's T.416 Information technology - Open Document Architecture (ODA) and interchange format: Character content architectures[36] which was adopted as ISO/IEC International Standard 8613-6 gives an alternative version that seems to be less supported.[dubious – discuss] The parameters after the '2', i.e. even the r,g,b are optional. Note that this is not just the above sequence with semicolon replaced by colon, there is a leading "colorspace ID" (this fact was missed by many terminal emulators, this omission seems to have come from KDE Konsole).[19] The definition of the colorspace ID is not included in that document so it may be blank to represent the unspecified default. As well as the '2' value to specify a Red-Green-Blue format (and the '5' above for a 0-255 indexed color) there are alternatives of '0' for implementation defined and '1' for transparent - neither of what have any further parameters; '3' specifies colors using a Cyan-Magenta-Yellow scheme, and '4' for a Cyan-Magenta-Yellow-Black one, the latter using the position marked as "unused" for the Black component:[citation needed]

ESC[ 38:2:⟨Color-Space-ID⟩:⟨r⟩:⟨g⟩:⟨b⟩:⟨unused⟩:⟨CS tolerance⟩:⟨Color-Space associated with tolerance: 0 for "CIELUV"; 1 for "CIELAB"⟩; m Select RGB foreground color
ESC[ 48:2:⟨Color-Space-ID⟩:⟨r⟩:⟨g⟩:⟨b⟩:⟨unused⟩:⟨CS tolerance⟩:⟨Color-Space associated with tolerance: 0 for "CIELUV"; 1 for "CIELAB"⟩; m Select RGB background color
```

Reply:
> In practice, this is not happening.

Have you noticed the check box in the Special Options tab (near the bottom) that has a big tool-tip trying to explain that it is needed to compensate for Game Servers that forget the Color Space ID parameter immediately after the `SGR 38 ; 2 ; ` following [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle) makes the code to parse these Esc sequences nasty and hard to verify. Please use the [*bona-fide*(*de-jure*) standard not the *de-facto*](https://stackoverflow.com/a/33206814/4805858) one...

Reply:
I'm tired of arguing with you. I'm changing it out of sheer exhaustion, but I don't have to like it.

Reply:
multiple standards exist, which we support, and I just don't have the energy to keep arguing about it. If it breaks later I expect you'll fix it.

Reply:
also, no, I haven't read the gigantic tooltip. As has been stated many times, the tooltips are out of hand and often aren't read because they're ridiculously long